tests/api/fuzzer.py

```diff
@@ -0,0 +1,319 @@
+
+
+
+class GetPath(object):
+    def __init__(self, url, spec):
+        self.url    = url
+        self.req_params = {}
+        self.opt_params = {}
+        self.success    = None
+        self.failures   = {}
+        if 'parameters' in spec.keys():
+            for p in spec['parameters']:
+                name = p['name']
+                req  = p.get('required',False)
+                target = self.req_params if req else self.opt_params
+                target[name] = Param(name, p['in'], p['type'])
+                if 'default' in p:
+                    defs = p['default']
+                    if isinstance(defs,list):
+                        for d in defs:
+                            target[name].values.add(d)
+                    else:
+                        target[name].values.add(defs)
+                if 'enum' in p:
+                    for v in p['enum']:
+                        target[name].values.add(v)
+                if req and len(target[name].values)==0:
+                    print(f"FAIL: No default values in swagger for required parameter {name} in {self.url}")
+            for code,schema in spec['responses'].items():
+                if code=="200" and 'schema' in schema:
+                    self.success = schema['schema']
+                elif code=="200":
+                    L.error(f"200 reponse with no schema in {self.url}")
+                else:
+                    self.failures[code] = schema
+
+    def generate_success(self, host):
+        args = "&".join([f"{p.name}={some(p.values)}" for p in self.req_params.values() ])
+        base_url = urllib.parse.urljoin(host,self.url)
+        test_url = f"{base_url}?{args}"
+        if url_filter.match(test_url):
+            print(f"TEST: {test_url}")
+            return requests.get(url=test_url)
+        L.debug(f"url_filter skips {test_url}")
+        #for p in self.req_params.values():
+        #    p.dump()
+        #for p in self.opt_params.values():
+        #    p.dump()
+
+    def generate_failure(self, host):
+        args = "&".join([f"{p.name}={some(p.values)}" for p in self.req_params.values() ])
+        base_url = urllib.parse.urljoin(host,self.url)
+        test_url = f"{base_url}?{args}"
+        print(f"TEST: {test_url}")
+        #for p in self.req_params.values():
+        #    p.dump()
+        #for p in self.opt_params.values():
+        #    p.dump()
+        return requests.get(url=test_url)
+
+    def validate(self, resp, expect_success):
+        try:
+            resp_json = json.loads(resp.text)
+        except json.decoder.JSONDecodeError as e:
+            L.error("Non-json response - how to validate?")
+            return
+        if resp.status_code==200:
+            if self.success is not None:
+                does_response_fit_schema(posixpath.join(self.url,"200"), self.success, resp_json)
+            else:
+                L.error("Missing schema?")
+            #print(json.loads(resp.text))
```

Comment:
Remove commented-out code please.

Reply:
It will all be gone by the time I remove the [WIP] marker (I'm still uncommenting it during dev for testing).